{
    "containers": {
        "cna": {
            "affected": [
                {
                    "product": "django-registration",
                    "vendor": "ubernostrum",
                    "versions": [
                        {
                            "status": "affected",
                            "version": "< 3.1.2"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "django-registration is a user registration package for Django. The django-registration package provides tools for implementing user-account registration flows in the Django web framework. In django-registration prior to 3.1.2, the base user-account registration view did not properly apply filters to sensitive data, with the result that sensitive data could be included in error reports rather than removed automatically by Django. Triggering this requires: A site is using django-registration < 3.1.2, The site has detailed error reports (such as Django's emailed error reports to site staff/developers) enabled and a server-side error (HTTP 5xx) occurs during an attempt by a user to register an account. Under these conditions, recipients of the detailed error report will see all submitted data from the account-registration attempt, which may include the user's proposed credentials (such as a password)."
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "HIGH",
                        "attackVector": "LOCAL",
                        "availabilityImpact": "NONE",
                        "baseScore": 3.7,
                        "baseSeverity": "LOW",
                        "confidentialityImpact": "LOW",
                        "integrityImpact": "LOW",
                        "privilegesRequired": "HIGH",
                        "scope": "CHANGED",
                        "userInteraction": "REQUIRED",
                        "vectorString": "CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:C/C:L/I:L/A:N",
                        "version": "3.1"
                    }
                }
            ],
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-209",
                            "description": "{\"CWE-209\":\"Generation of Error Message Containing Sensitive Information\"}",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "dateUpdated": "2021-04-01T21:15:14",
                "orgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
                "shortName": "GitHub_M"
            },
            "references": [
                {
                    "tags": [
                        "x_refsource_CONFIRM"
                    ],
                    "url": "https://github.com/ubernostrum/django-registration/security/advisories/GHSA-58c7-px5v-82hh"
                }
            ],
            "source": {
                "advisory": "GHSA-58c7-px5v-82hh",
                "discovery": "UNKNOWN"
            },
            "title": "Potential sensitive information disclosed in error reports",
            "x_legacyV4Record": {
                "CVE_data_meta": {
                    "ASSIGNER": "security-advisories@github.com",
                    "ID": "CVE-2021-21416",
                    "STATE": "PUBLIC",
                    "TITLE": "Potential sensitive information disclosed in error reports"
                },
                "affects": {
                    "vendor": {
                        "vendor_data": [
                            {
                                "product": {
                                    "product_data": [
                                        {
                                            "product_name": "django-registration",
                                            "version": {
                                                "version_data": [
                                                    {
                                                        "version_value": "< 3.1.2"
                                                    }
                                                ]
                                            }
                                        }
                                    ]
                                },
                                "vendor_name": "ubernostrum"
                            }
                        ]
                    }
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "eng",
                            "value": "django-registration is a user registration package for Django. The django-registration package provides tools for implementing user-account registration flows in the Django web framework. In django-registration prior to 3.1.2, the base user-account registration view did not properly apply filters to sensitive data, with the result that sensitive data could be included in error reports rather than removed automatically by Django. Triggering this requires: A site is using django-registration < 3.1.2, The site has detailed error reports (such as Django's emailed error reports to site staff/developers) enabled and a server-side error (HTTP 5xx) occurs during an attempt by a user to register an account. Under these conditions, recipients of the detailed error report will see all submitted data from the account-registration attempt, which may include the user's proposed credentials (such as a password)."
                        }
                    ]
                },
                "impact": {
                    "cvss": {
                        "attackComplexity": "HIGH",
                        "attackVector": "LOCAL",
                        "availabilityImpact": "NONE",
                        "baseScore": 3.7,
                        "baseSeverity": "LOW",
                        "confidentialityImpact": "LOW",
                        "integrityImpact": "LOW",
                        "privilegesRequired": "HIGH",
                        "scope": "CHANGED",
                        "userInteraction": "REQUIRED",
                        "vectorString": "CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:C/C:L/I:L/A:N",
                        "version": "3.1"
                    }
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "eng",
                                    "value": "{\"CWE-209\":\"Generation of Error Message Containing Sensitive Information\"}"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://github.com/ubernostrum/django-registration/security/advisories/GHSA-58c7-px5v-82hh",
                            "refsource": "CONFIRM",
                            "url": "https://github.com/ubernostrum/django-registration/security/advisories/GHSA-58c7-px5v-82hh"
                        }
                    ]
                },
                "source": {
                    "advisory": "GHSA-58c7-px5v-82hh",
                    "discovery": "UNKNOWN"
                }
            }
        }
    },
    "cveMetadata": {
        "assignerOrgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
        "assignerShortName": "GitHub_M",
        "cveId": "CVE-2021-21416",
        "datePublished": "2021-04-01T21:15:14",
        "dateReserved": "2020-12-22T00:00:00",
        "dateUpdated": "2021-04-01T21:15:14",
        "state": "PUBLISHED"
    },
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0"
}